### PR TITLE
Rwa 1000 - Insert DLQ message into DB

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-1000

### Change description ###

 Insert DLQ message into DB

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerEndpointTest.java [hunk 1]
```diff
@@ -224,8 +230,21 @@ void dlq_case_event_message_should_be_stored_and_return_200_ok() throws Exceptio
...
+        void dlq_case_event_message_should_not_be_stored_and_return_200_ok_when_feature_flag_disabled()
+                throws Exception {
 
+            when(launchDarklyFeatureFlagProvider.getBooleanValue(any(), any())).thenReturn(false);
+            String messageId = randomMessageId();
+
+            MvcResult result = postMessage(messageId, status().isCreated(), true);
+
+            String content = result.getResponse().getContentAsString();
+            assertEquals(201, result.getResponse().getStatus(), content);
```
Comment (alexbaeza) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerEndpointTest.java:242]: Are we missing an assertion here to ensure the message is not stored?

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java [hunk 2]
```diff
@@ -47,7 +47,13 @@ public EventMessageReceiverService(ObjectMapper objectMapper,
...
     public CaseEventMessage handleDlqMessage(String messageId, String message) {
         log.info("Received Case Event Dead Letter Queue message with id '{}'", messageId);
-        return handleMessage(messageId, message, true);
+        if (featureFlagProvider.getBooleanValue(DLQ_DB_INSERT, getUserId(message))) {
+            return handleMessage(messageId, message, true);
+        } else {
+            log.info("Feature flag '{}' evaluated to false. Message not inserted into database",
+                    DLQ_DB_INSERT.getKey());
+        }
+        return null;
```
Comment (alexbaeza) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java:56]: Just did a quick search, but question: Do we use this  CaseEventMessage or null that we return?

Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java:56]: It is used by the endpoint in CaseEventHandlerController.java used to emulate receiving a message from the consumer

public CaseEventMessage caseEventHandler(@Valid @RequestBody String message,
                                             @PathVariable("message_id") final String messageId,
                                             @RequestParam(value = "from_dlq",
                                                 required = false) final Boolean fromDlq) 
                                                 
It is used for the assertions in CaseEventHandlerControllerTest

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerEndpointTest.java (hunk 1 above)
Comment (sullivangareth) [reply src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerEndpointTest.java:244]: Fixed (assertion added) in f4a2147

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java (hunk 2 above)
Comment (alexbaeza) [reply src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java:56]: Thanks :)

**Review by alexbaeza — DISMISSED** [review]

**Review by rafalkalita — DISMISSED** [review]
LGTM

**Review by BenNzewi — APPROVED** [review]

**Pull request merged**